env_test.go

```diff
@@ -643,6 +657,15 @@ func TestErrorRequiredNotSet(t *testing.T) {
 	assert.EqualError(t, Parse(cfg), "env: required environment variable \"IS_REQUIRED\" is not set")
 }
 
+func TestErrorRequiredNotSetWithDefault(t *testing.T) {
+	type config struct {
+		IsRequired string `env:"IS_REQUIRED,required" envDefault:"important"`
+	}
+
+	cfg := &config{}
+	assert.EqualError(t, Parse(cfg), "env: required environment variable \"IS_REQUIRED\" is not set")
```

Comment:
was this the behavior before?

I would not expect it to error 🤔

Reply:
Yes, it seem to have been. I wrote the test with the intention to preserve this behavior since it seem to have been there but not tested. 

https://github.com/caarlos0/env/blob/master/env.go#L185
Returns an error if the environment variable is not present, regardless of a default existing or not. 

In my mind this makes sens, since a envDefault would render the require annotation meaningless.